### PR TITLE
Connection browser enhancements

### PR DESCRIPTION
**Pull request by chrjorgensen**
### Changes

This PR adds some enhancements to the server browser:
- add a sort connection option 

image

- add Rename Connection menu item

image

- remove most connection commands from command palette
- only show Connection Settings in command palette when connected
- remove cached server settings when option is changed to disable

### Checklist

* [x] have tested my change
* [x] Remove any/all `console.log`s I added
* [x] eslint is not complaining

### CODE REVIEW

File: src/locale/ids/fr.ts [hunk 1]
```diff
@@ -45,6 +45,9 @@ export const fr: Locale = {
   'connectionBrowser.connectTo.lastUsed': `Dernière utilisation: {0}`,
   'connectionBrowser.connectTo.title': `Dernières connexions IBM i`,
   'connectionBrowser.connectTo.error': `Utilisez l'Explorateur de Serveur pour sélectionner le serveur auquel se connecter.`,
+  'connectionBrowser.renameConnection.prompt': `Rename connection {0}`,
```
Comment (chrjorgensen) [inline src/locale/ids/fr.ts:48]: @sebjulliand Could I ask you to translate these three new texts into French? :pray:

**Comment by worksofliam** [comment]
Not part of my review but an issue nonetheless:

If I rename a connection, enter a new name, and get this error:

image

But then if I refresh the view, the name is updated correctly.

**Comment by chrjorgensen** [comment]
@worksofliam Could your error be caused by some "molested" configuration settings in VS Code, where the `name` attribute is not there as expected?

I've seen many examples of this in my own settings - like the `"button" = "save"`example, which is also seen in other users settings in some of the issues...

A connection has values in up to four different places in the settings / storage, and currently there is a danger in the code, that only part of them gets the new name for the connection. So be careful when testing - do not test on important connections!

I will add a try-catch block to catch any errors and not update anything if errors occurred - stay tuned...

**Comment by chrjorgensen** [comment]
@worksofliam Error handling has been added and you should get more error information - and not get any connection changed if error occurred. (Tested this with `settings.json` opened - cool to see the file being updated by the rename command!)

@sebjulliand Now even more messages to translate to French... 😉

File: src/views/ConnectionBrowser.ts [hunk 2]
```diff
@@ -75,6 +75,70 @@ export class ObjectBrowserProvider {
...
+
+              // Then get the password key
+              const secret = await context.secrets.get(`${server.name}_password`);
+
+              // No errors - update the settings.
+              await GlobalConfiguration.set(`connectionSettings`, connectionSettings);
+              await GlobalConfiguration.set(`connections`, connections);
+              if(cachedConnectionSettings) {
+              GlobalStorage.get().setServerSettingsCache(newName, cachedConnectionSettings);
+              GlobalStorage.get().deleteServerSettingsCache(server.name);
```
Comment (sebjulliand) [inline src/views/ConnectionBrowser.ts:119]: Indentation
```suggestion
                GlobalStorage.get().setServerSettingsCache(newName, cachedConnectionSettings);
                GlobalStorage.get().deleteServerSettingsCache(server.name);
```

File: src/views/ConnectionBrowser.ts [hunk 3]
```diff
@@ -75,6 +75,70 @@ export class ObjectBrowserProvider {
...
+            } catch (e: any) {
+              vscode.window.showErrorMessage(
+                t(`connectionBrowser.renameConnection.error`, server.name, e.message || String(e)));
+            }
+          }
+        }
+      }),
+
+      vscode.commands.registerCommand(`code-for-ibmi.sortConnections`, async () => {
+        let connections = GlobalConfiguration.get<ConnectionData[]>(`connections`) || [];
```
Comment (sebjulliand) [inline src/views/ConnectionBrowser.ts:136]: Use `const` whenever you can
```suggestion
        const connections = GlobalConfiguration.get<ConnectionData[]>(`connections`) || [];
```

**Review by sebjulliand — CHANGES_REQUESTED** [review]
Looks good and works fine so far! Thank you!
Just some very minor style change were requested, but that's just me being me.

File: src/views/ConnectionBrowser.ts (hunk 3 above)
Comment (chrjorgensen) [reply src/views/ConnectionBrowser.ts:136]: Agreed - just me getting confused that you can change a `const`... :roll_eyes:

Comment (sebjulliand) [reply src/views/ConnectionBrowser.ts]: The object state can change and that's fine.It's reference cannot, giving you the assurance that you'll always manipulate the same object and it won't be replaced somehow.

**Review by sebjulliand — APPROVED** [review]
Brilliant! Let's merge!

**Pull request merged**